cli/src/commands/git/push.rs

```diff
@@ -746,6 +768,63 @@ fn classify_bookmark_update(
     }
 }
 
+/// Creates a bookmark for a single `--named` argument and returns its name
+///
+/// The logic is not identical to that of `jj bookmark create` since we need to
+/// make sure the new bookmark is safe to push.
+fn create_explicitly_named_bookmarks(
+    ui: &mut Ui,
+    tx: &mut WorkspaceCommandTransaction<'_>,
+    name_revision: &String,
+) -> Result<RefNameBuf, CommandError> {
+    let hint = "For example, `--named myfeature=@` is valid syntax";
+    let Some((name_str, revision_str)) = name_revision.split_once('=') else {
+        return Err(cli_error(format!(
+            "Argument '{name_revision}' must include '=' and have the form NAME=REVISION"
+        ))
+        .hinted(hint));
+    };
+    if name_str.is_empty() || revision_str.is_empty() {
+        return Err(cli_error(format!(
+            "Argument '{name_revision}' must have the form NAME=REVISION, with both NAME and \
+             REVISION non-empty."
+        ))
+        .hinted(hint));
+    }
+    let name = parse_bookmark_name(name_str).map_err(|err| {
+        cli_error_with_message(
+            format!("Could not parse '{name_str}' as a bookmark name."),
+            err,
+        )
+        .hinted(hint)
+    })?;
+    let symbol = name.as_symbol();
+    if tx.repo().view().get_local_bookmark(&name).is_present() {
+        return Err(user_error_with_hint(
+            format!("Bookmark already exists: '{symbol}'"),
+            format!(
+                "Use 'jj bookmark move' to move it, and 'jj git push -b {symbol} [--allow-new]' \
+                 to push it.",
+            ),
+        ));
+    }
+    if has_tracked_remote_bookmarks(tx.repo().view(), &name) {
+        return Err(user_error_with_hint(
+            format!("Tracked remote bookmarks exist for deleted bookmark: {symbol}",),
```

Comment:
nit: remove redundant `,`

Reply:
Again, surprised neither `cargo fmt` nor clippy caught this. I guess it's a macro.